x/marker/types/marker_test.go

```diff
@@ -406,3 +406,69 @@ func TestRemovesFromRequiredAttributes(t *testing.T) {
 		})
 	}
 }
+
+func TestNetAssetValueConstructor(t *testing.T) {
+	price := sdk.NewInt64Coin("jackthecat", 406)
+	volume := uint64(100)
+	actual := NewNetAssetValue(price, volume)
+	assert.Equal(t, price, actual.Price)
+	assert.Equal(t, volume, actual.Volume)
+	assert.Equal(t, uint64(0), actual.UpdatedBlockHeight, "update time should not be set")
+}
+
+func TestNetAssetValueValidate(t *testing.T) {
+	tests := []struct {
+		name   string
+		nav    NetAssetValue
+		expErr string
+	}{
+		{
+			name: "invalid denom",
+			nav: NetAssetValue{
+				Volume: 406,
+			},
+			expErr: "invalid denom: ",
+		},
+		{
+			name: "volume is not positive",
+			nav: NetAssetValue{
+				Price:  sdk.NewInt64Coin("jackthecat", 420),
+				Volume: 0,
+			},
+			expErr: "marker net asset value volume must be positive value",
+		},
+		{
+			name: "volume must be positive if value is greater than 1",
+			nav: NetAssetValue{
+				Price:  sdk.NewInt64Coin("usdcents", 1),
+				Volume: 0,
+			},
+			expErr: "marker net asset value volume must be positive value",
+		},
+		{
+			name: "successful with 0 volume and coin",
```

Comment:
Does this make sense to support? What value is there for a zero volume/zero price entry?